### DISCUSSION TITLE
How to change RightPanel spacing for schema imports only?

### DISCUSSION THREAD

**Discussion by erodewald** [rohit-gohri/redocusaurus#165] [Q&A]
The default theme of redoc has a lot of spacing built in, but when I include a schema import _with_ the `example` enabled, there's too little horizontal space, which results in a lot of scrolling in my example. 

I was hoping that I could pass in some additional `props` into the React component, but it seems to have no effect. 

```typescript
<ApiSchema
	id="diagnostics"
	example
	pointer="#/components/schemas/DiagnosticLifecycleEvent"
	exampleRef="#/components/examples/simulatedLifecycleEvent"
        theme={{spacing:{unit:0}}}                                   <!-- attempt 1 -->
        options={{theme:{spacing:{unit:0,sectionHorizontal:20}}}}    <!-- attempt 2 -->
/>

```

I'm using the following line as a clue that this is possible:

https://github.com/rohit-gohri/redocusaurus/blob/c5e55d56a025e04a57fd16e25f7fddd5010e6f51/packages/docusaurus-theme-redoc/src/theme/ApiSchema/ApiSchema.tsx#L42

Along with this from redocly, which seems to correlate: 
https://github.com/Redocly/redoc/blob/a160498089a44ebf540745b55fbaf18a54eb9d88/src/components/SchemaDefinition/SchemaDefinition.tsx?ts=2#L86

I am probably way off, let me know if you have any ideas.

**Comment by rohit-gohri** [comment]
The theme is coming from the `store` and that is using the global theme. If you update theme options in docusaurus.config does it work then? If it does then we can think of separating the default themes for schema import and full docs.

**Reply by erodewald** [reply]
Yes, when I modify the redocusaurus'  `theme.theme.spacing.sectionHorizontal`, it is modified globally.

**Reply by rohit-gohri** [reply]
Cool. I'll try to create separate settings/prop for Schema Imports
